src/msg/Messenger.cc

```diff
@@ -22,10 +22,12 @@ Messenger *Messenger::create(CephContext *cct, const string &type,
 			     entity_name_t name, string lname,
 			     uint64_t nonce, uint64_t features)
 {
-  int r = -1;
-  srand(time(NULL));
-  if (type == "random")
+  static int r = -1;
```

Comment:
could use a dedicated `std::once_flag` and `std::call_once()` here? not sure if this is overkill though.